ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -325,40 +379,57 @@ module Json = struct
     in
-    List.map
-      (fun msg ->
-        let params = params_in_msg msg |> of_params in
-        let base_assoc_list =
-          [
-            ("method_name", `String msg.msg_name)
-          ; ("class_name", `String obj.name)
-          ; ("class_name_exported", `String (snake_to_camel obj.name))
-          ; ("method_name_exported", `String (snake_to_camel msg.msg_name))
-          ; ("description", desc_of_msg msg ctor_fields)
-          ; ("result", of_result obj msg)
-          ; ("params", params)
-          ; ("errors", of_errors msg.msg_errors)
-          ; ("has_error", `Bool (msg.msg_errors <> []))
-          ; ("async", `Bool msg.msg_async)
-          ]
-        in
-        (* Since the param of `session *Session` isn't needed in functions of session object,
-           we add a special "func_params" field for session object to ignore `session *Session`.*)
-        if obj.name = "session" then
-          `O
-            (("func_params", msg.msg_params |> of_params)
-            :: (add_session_info obj.name msg.msg_name @ base_assoc_list)
-            )
-        else
-          `O base_assoc_list
-      )
-      obj.messages
+    obj.messages
+    |> List.rev_map (fun msg ->
+           let of_message (num, params, rel_version) =
+             let base_assoc_list =
+               [
+                 ("method_name", `String msg.msg_name)
+               ; ("class_name", `String obj.name)
+               ; ("class_name_exported", `String (snake_to_camel obj.name))
+               ; ( "method_name_exported"
+                 , `String (method_name_exported obj msg num)
+                 )
+               ; ("description", desc_of_msg msg ctor_fields)
+               ; ("result", of_result obj msg)
+               ; ("params", `A (of_params params))
+               ; ("errors", of_errors msg.msg_errors)
+               ; ("has_error", `Bool (msg.msg_errors <> []))
+               ; ("async", `Bool msg.msg_async)
+               ; ("version", `String rel_version)
+               ]
+             in
+             (* Since the param of `session *Session` isn't needed in functions of session object,
+                we add a special "func_params" field for session object to ignore `session *Session`.*)
+             match (obj.name, msg.msg_session) with
+             | "session", true ->
+                 `O
+                   (("func_params", `A (params |> List.tl |> of_params))
+                   :: (add_session_info obj.name msg.msg_name @ base_assoc_list)
+                   )
+             | "session", false ->
+                 `O
+                   (("func_params", `A (of_params params))
+                   :: (add_session_info obj.name msg.msg_name @ base_assoc_list)
+                   )
+             | _ ->
+                 `O base_assoc_list
+           in
+           msg |> group_params |> List.map of_message
+       )
+    |> List.concat
+    |> Listext.List.setify
```

Comment:
Why we need this? It's supposed to be a set already. Otherwise, something may be wrong.

Reply:
There is a bug here. Fixed. Thanks.